soh/src/code/z_play.c

```diff
@@ -271,7 +271,11 @@ void Gameplay_Init(GameState* thisx) {
     u8 tempSetupIndex;
     s32 pad[2];
 
-    if (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SKIP_CHILD_STEALTH)) {
+    // Skip Child Stealth when option is enabled, Zelda's Letter isn't obtained and Impa's reward hasn't been received
+    // eventChkInf[4] & 1 = Got Zelda's Letter
+    // eventChkInf[5] & 0x200 = Got Impa's reward
+    if (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SKIP_CHILD_STEALTH) &&
+        !(gSaveContext.eventChkInf[4] & 1) && !(gSaveContext.eventChkInf[5] & 0x200)) {
```

Comment:
might be nice to add a little more context to the rest of this if block via comments explaining what's happening

all of this makes me think it'd be awesome to replace all the entrance index magic numbers with nice names in an enum or something but for now just adding some comments here would go a long way

Reply:
I'm going to be honest, I don't really know what the entrance values here are either, I just changed the conditional to trigger them. Do you happen to remember how this part worked, or do you want me to dive in and find out?

Reply:
I don't remember off the top of my head, I can look into it later if you don't have time to dive in